src/daft-distributed/src/program/collect.rs

```diff
@@ -0,0 +1,54 @@
+use std::collections::HashMap;
+
+use common_error::DaftResult;
+use common_partitioning::PartitionRef;
+use daft_local_plan::LocalPhysicalPlanRef;
+
+use super::{ProgramContext, RunningProgram};
+use crate::{
+    channel::{create_channel, Sender},
+    scheduling::dispatcher::TaskDispatcherHandle,
+    stage::StageContext,
+};
+
+pub(crate) struct CollectProgram {
+    program_context: ProgramContext,
+}
+
+impl CollectProgram {
```

Comment:
this should ideally be implemented in a Program Trait right?

Reply:
we can ideally follow a tree structure then where nodes can hold onto `Box<dyn Program>`